editor/editor_settings.cpp

```diff
@@ -216,17 +216,83 @@ Variant _EDITOR_DEF(const String &p_var, const Variant &p_default) {
 	return p_default;
 }
 
+static Dictionary _get_builtin_script_templates() {
+	Dictionary templates;
+
+	//No Comments
+	templates["no_comments.gd"] =
+			"extends %BASE%\n"
+			"\n"
+			"func _ready():\n"
+			"%TS%pass\n";
+
+	//Empty
+	templates["empty.gd"] =
+			"extends %BASE%"
+			"\n"
+			"\n";
+
+	//Default Extended
+	templates["default_extended.gd"] =
+			"extends %BASE%\n"
+			"# ctrl+click or cmd+click to get information about the Node type this script is attached to.\n"
+			"# any variable and function is available in the scope of this script.\n"
+			"# don't forget to also checkout the classes this node Inherits from! All those functions + parameters are available too.\n"
+			"\n"
+			"\n"
+			"func _ready():\n"
+			"	# if you dont need one of the callbacks set it to false: set_process(false)"
+			"\n"
+			"	set_process(true) 		# This tells godot to call the '_ready' callback  on this node each frame\n"
+			"	set_process_input(true) # This tells godot to call the '_input' callback on this node for each input event\n"
+			"	set_fixed_process(true) # This tells godot to call the '_fixed_process' callback on this node each PHYSICS frame\n"
+			"\n"
+			"	# THIS SECTION WILL ONLY WORK WHEN CREATING THE APPROPIATE NODES AND ATTACH SCRIPTS WITH THE CORRECT FUNCTIONS\n"
+			"	# Comment this section or add the needed nodes and adapt the names\n"
+			"	var my_node = $child_node_name # Accessing a child node and storing it in a variable. THERE HAS TO BE A CHILD NODE WITH THE NAME: 'child_node_name'\n"
+			"	var a_child_from_a_child = $child_node_name/sprite # Use '/' to access any node in the child tree from the current node.\n"
+			"	my_node.position = Vector2(100,100) # You can set any builtin parameter or call any function of the node you accessed.\n"
+			"	a_child_from_a_child.a_coustom_function(name) # You can call any function defined in the script attached to my_node just like that.\n"
+			"\n"
+			"_process(delta): \n"
+			"	# Anything that needs to be calculated each frame goes here.\n"
+			"	pass\n"
+			"\n"
+			"func _fixed_process(delta):\n"
+			"	#this function is called each physics frame. And synced with the physics engine\n"
+			"	#the physics might be calculated in a different thread so interacting with it in _process might result in unwanted behaviour\n"
+			"	pass\n"
+			"\n"
+			"func _input(event):\n"
+			"	#the event variable can be of any Type that extends from 'InputEvent'"
+			"	if event is InputEventMouseButton: #Use the 'is' statement to check the type of the input event\n"
+			"		if event.button_index == BUTTON_LEFT: #each input event type has differnt variables. InputEventMouseButton has the parameter button_index.\n"
+			"			print('BUTTON_LEFT')\n"
+			"	if event is InputEventKey: #To find the available parameters for a specific event: check the help page with: ctr+click on InputEventKey\n"
+			"		event.scancode == KEY_SPACE && event.pressed:\n"
+			"		print('KEY_SPACE')\n";
```

Comment:
Also, please use %TS% for indentation, so it respects user settings regarding tabs vs spaces.